docs/source/api/data-management/graphs.rst

```diff
@@ -0,0 +1,260 @@
+.. ******************************************************************************
+.. * Copyright 2020-2021 Intel Corporation
+.. *
+.. * Licensed under the Apache License, Version 2.0 (the "License");
+.. * you may not use this file except in compliance with the License.
+.. * You may obtain a copy of the License at
+.. *
+.. *     http://www.apache.org/licenses/LICENSE-2.0
+.. *
+.. * Unless required by applicable law or agreed to in writing, software
+.. * distributed under the License is distributed on an "AS IS" BASIS,
+.. * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+.. * See the License for the specific language governing permissions and
+.. * limitations under the License.
+.. *******************************************************************************/
+
+.. highlight:: cpp
+
+.. _api_graphs:
+
+======
+Graphs
+======
+
+Refer to :ref:`Developer Guide: Graphs <dm_graphs>`.
+
+.. _graph_programming_interface:
+
+---------------------
+Programming interface
+---------------------
+
+All types and functions in this section are declared in the
+``oneapi::dal::preview`` namespace and are available via inclusion of the
+``oneapi/dal/graph/common.hpp`` header file.
+
+Graph
+-----
+
+The :txtref:`graph` concept is represented by the types with the ``_graph`` suffix and all of them are
+:term:`reference-counted <Reference-counted object>`:
+
+1. The instance stores pointers to the graph :capterm:`topology`
+   and :capterm:`attributes <Attribute>` of vertices and edges.
+
+2. The reference count indicating how many graph objects refer to the same implementation.
+
+3. The graph increments the reference count
+   for it to be equal to the number of graph objects sharing the same implementation.
+
+4. The graph decrements the reference count when the
+   graph goes out of the scope. If the reference count is zero, the graph
+   frees its implementation.
+
+.. _api_graph_template_type:
+
+The :txtref:`graph` types are defined as templated classes:
+
+.. code-block:: cpp
+
+  template <typename VertexValue,
+            typename EdgeValue,
+            typename GraphValue,
+            typename IndexType,
+            typename Allocator>
+  class [graph_name]_graph;
+
+.. .. namespace:: oneapi::dal::preview
+.. .. struct:: graph_traits
```

Comment:
I've left comments on similar lines: is this supposed to be here? these lines are commented out and are not showing in the output

Reply:
Got it. I thought you've left comments for each similar line

Reply:
```suggestion
```